src/StackExchange.Redis/ConfigurationOptions.cs

```diff
@@ -296,24 +297,64 @@ private static RemoteCertificateValidationCallback TrustIssuerCallback(X509Certi
+                }
+                // If we're not valid due to chain errors - check against the trusted issuer
+                // Note that we're only proceeding at all here if the *only* issue is chain errors (not more flags in SslPolicyErrors)
+                return sslPolicyError == SslPolicyErrors.RemoteCertificateChainErrors
+                       && certificate is X509Certificate2 v2
+                       && CheckTrustedIssuer(v2, certificateChain, issuer);
+            };
         }
 
-        private static bool CheckTrustedIssuer(X509Certificate2 certificateToValidate, X509Certificate2 authority)
+        private static bool CheckTrustedIssuer(X509Certificate2 certificateToValidate, X509Chain? chainToValidate, X509Certificate2 authority)
         {
-            // reference: https://stackoverflow.com/questions/6497040/how-do-i-validate-that-a-certificate-was-created-by-a-particular-certification-a
-            X509Chain chain = new X509Chain();
+            // Reference:
+            // https://stackoverflow.com/questions/6497040/how-do-i-validate-that-a-certificate-was-created-by-a-particular-certification-a
+            // https://github.com/stewartadam/dotnet-x509-certificate-verification
+            using X509Chain chain = new X509Chain();
             chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
-            chain.ChainPolicy.RevocationFlag = X509RevocationFlag.ExcludeRoot;
             chain.ChainPolicy.VerificationFlags = X509VerificationFlags.AllowUnknownCertificateAuthority;
-            chain.ChainPolicy.VerificationTime = DateTime.Now;
+            chain.ChainPolicy.VerificationTime = chainToValidate?.ChainPolicy?.VerificationTime ?? DateTime.Now;
             chain.ChainPolicy.UrlRetrievalTimeout = new TimeSpan(0, 0, 0);
 
             chain.ChainPolicy.ExtraStore.Add(authority);
-            return chain.Build(certificateToValidate);
+            // This only verifies that the chain is valid, but with AllowUnknownCertificateAuthority could trust
+            // self-signed or partial chained vertificates
+            try
+            {
+                var chainIsVerified = chain.Build(certificateToValidate);
+                if (chainIsVerified)
+                {
+                    // Our method is "TrustIssuer", which means any intermediate cert we're being told to trust
+                    // is a valid thing to trust, up until it's a root CA
+                    bool found = false;
+                    foreach (var chainElement in chain.ChainElements)
+                    {
+#if NET8_0_OR_GREATER
+#error TODO: use RawDataMemory (needs testing)
+#endif
+                        using var chainCert = chainElement.Certificate;
+                        if (!found && chainCert.RawData.SequenceEqual(authority.RawData))
+                        {
+                            found = true;
+                        }
+                    }
+                    return found;
```

Comment:
I think I saw Kevin mention it already; but you could lift `authority.RawData` out of the loop

```suggestion
                    bool found = false;
                    byte[] authorityData = authority.RawData;
                    foreach (var chainElement in chain.ChainElements)
                    {
#if NET8_0_OR_GREATER
#error TODO: use RawDataMemory (needs testing)
#endif
                        using var chainCert = chainElement.Certificate;
                        if (!found && chainCert.RawData.SequenceEqual(authorityData))
                        {
                            found = true;
                        }
                    }
                    return found;
```

to reduce the number of defensive copies of that array